src/extension/loadLogs.ts

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+/// <reference path="jsonSourceMap.d.ts" />
+import { execFileSync } from 'child_process';
+import * as fs from 'fs';
+import jsonMap from 'json-source-map';
+import { join } from 'path';
+import { Log } from 'sarif';
+import { eq, gt, lt } from 'semver';
+import { tmpNameSync } from 'tmp';
+import { ProgressLocation, Uri, window } from 'vscode';
+import { augmentLog, JsonMap } from '../shared';
+import { Store } from './store';
+
+export async function loadLogs(uris: Uri[], token?: { isCancellationRequested: boolean }) {
+    const logs = uris
+        .map(uri => {
+            if (token?.isCancellationRequested) return undefined;
+            try {
+                const file = fs.readFileSync(uri.fsPath, 'utf8')  // Assume scheme file.
+                    .replace(/^\uFEFF/, ''); // Trim BOM.
+                const {data: log, pointers} = jsonMap.parse(file) as { data: Log, pointers: JsonMap};
+                log._uri = uri.toString();
+                log._jsonMap = pointers;
+                return log;
+            } catch (error) {
+                window.showErrorMessage(`Failed to parse '${uri.fsPath}'`);
+                return undefined;
+            }
+        })
+        .filter(log => log) as Log[];
+    const logsNoUpgrade = [] as Log[];
+    const logsToUpgrade = [] as Log[];
+    const warnUpgradeExtension = logs.some(log => detectUpgrade(log, logsNoUpgrade, logsToUpgrade));
+    const upgrades = logsToUpgrade.length;
+    if (upgrades) {
+        await window.withProgress(
+            { location: ProgressLocation.Notification },
+            async progress => {
+                for (const [i, oldLog] of logsToUpgrade.entries()) {
+                    if (token?.isCancellationRequested) break;
+                    progress.report({
+                        message: `Upgrading ${i + 1} of ${upgrades} log${upgrades === 1 ? '' : 's'}...`,
+                        increment: 1 / upgrades * 100
```

Comment:
>1 [](start = 35, length = 1)

i?

Reply:
Their API wants the increment/delta as supposed to the total. I found it confusing too.

---
In reply to: [441211593](https://github.com/microsoft/sarif-vscode-extension/pull/282#discussion_r441211593) [](ancestors = 441211593)